app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt

```diff
@@ -66,13 +103,33 @@ fun AddTripScreen(
   val formattedStartDate = startDate?.let { dateFormat.format(Date(it)) } ?: ""
   val formattedEndDate = endDate?.let { dateFormat.format(Date(it)) } ?: ""
 
-  val disabledBorderColor = Color.DarkGray
-  val disabledTextColor = Color.Black
-
+  val permissionVersion =
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        READ_MEDIA_IMAGES
+      } else {
+        READ_EXTERNAL_STORAGE
+      }
   val galleryLauncher =
       rememberLauncherForActivityResult(
           contract = ActivityResultContracts.GetContent(),
           onResult = { uri -> uri?.let { imageUri = it.toString() } })
+  val permissionLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+        if (isGranted) {
+          galleryLauncher.launch("image/*")
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE &&
+            ContextCompat.checkSelfPermission(context, READ_MEDIA_VISUAL_USER_SELECTED) ==
+                PERMISSION_GRANTED) { // This checks in case the user provided only limited access
+          // to their gallery
+          galleryLauncher.launch("image/*")
```

Comment:
I understand the approach you've taken to handle the permission checks, and the implementation correctly addresses the various scenarios across different Android SDK versions. While the current implementation is effective, there might be opportunities to streamline the logic for better readability and maintainability. For example, you could use the checkPermission function.

Reply:
You're right, I'll modularise this a bit more. Thanks!